routes/bridges/gassistant/index.js

```diff
@@ -0,0 +1,235 @@
+app.intent('actions.intent.MAIN', async (conv) => {
+
+    const [anonymous, user] = await retrieveUser(conv.body.user.accessToken);
+
+    const locale = conv.body.user.locale;
+    const conversationId = conv.body.conversation.conversationId;
+    const assistantUser = { name: conv.user.name.display || 'User', isOwner: true };
+    const delegate = new GoogleAssistantDelegate(locale);
+
+    return Q.try(() => {
+        return EngineManager.get().getEngine(user.id);
+    }).then((engine) => {
+        return engine.assistant.getOrOpenConversation('google_assistant:' + conversationId,
+            assistantUser, delegate, { anonymous, showWelcome: true, debug: true });
+    }).then(() => {
+        // Send welcome message
+        delegate._buffer.forEach((reply) => conv.ask(reply));
+    });
+});
+
+// Immediate response after user authenticates
+app.intent('actions.intent.SIGN_IN', (conv, input, signin) => {
+    if (signin.status === 'OK')
+        conv.ask("Thank you for signing in! What would you like to do next?");
+    else
+        conv.ask("You were unable to log in. Is there something else you want to do?");
+});
+
+// All other responses
+app.intent('actions.intent.TEXT', async (conv, input) => {
+    // Quick hack so that Almond recognizes bye and goodbye
+    // and returns user to Google Assistant
+    if (input === 'bye' || input === 'goodbye')
+        return conv.close("See you later!");
+    // TODO - better way for user to initiate sign in
+    if (input === 'I want to sign in')
+        // This will reply "<To get your account details>, I need to link your
+        // <action> account to Google. Is that okay?"
+        // Answering "yes" will generate the log-in link
+        return conv.ask(new SignIn("To get your account details"));
+
+    const [anonymous, user] = await retrieveUser(conv.body.user.accessToken);
+
+    const locale = conv.body.user.locale;
+    const conversationId = conv.body.conversation.conversationId;
+    const assistantUser = { name: conv.user.name.display || 'User', isOwner: true };
+    const delegate = new GoogleAssistantDelegate(locale);
+
+    return Q.try(() => {
+        return EngineManager.get().getEngine(user.id);
+    }).then((engine) => {
+        return engine.assistant.getOrOpenConversation('google_assistant:' + conversationId,
+            assistantUser, delegate, { anonymous, showWelcome: false, debug: true });
+    }).then((conversation) => {
+        if (input.startsWith('\\t'))
+            return conversation.handleThingTalk(input.substring(3));
+        else
+            return conversation.handleCommand(input);
+    }).then(() => {
+        if (delegate._buffer) {
+            delegate._buffer.forEach((reply) => conv.ask(reply));
+            // Another way to initiate authentication, initiated by Almond
+            if (delegate._requestSignin)
+                conv.ask(new SignIn("To get your account details"));
+        } else {
+            conv.close("Consider it done.");
```

Comment:
Did you find a case where Almond replies nothing? For things that don't return a result, Almond should always reply something.
(The two exceptions are "stop", which intentionally shushes Almond, and "wakeup", which does absolutely nothing.)

Reply:
Okay for "shush", because Almond replies nothing, if we don't do `conv.close("Consider it done.")`, then Google Assistant will reply "Sorry, SmartNews isn't responding right now. Please try again soon."
So should I just keep the `conv.close`?

Reply:
I don't know what are the rules around Google Assistant to be honest. We should probably always return something though, it kind of makes sense. So this is fine.